### PR TITLE
Add unit tests for com.vip.vjtools.vjtop.util.Formats

### PR DESCRIPTION
**Pull request by petersear**
Hi
I've written some more unit tests, with the help of [Diffblue Cover](https://www.diffblue.com/opensource), for `com.vip.vjtools.vjtop.util.Formats`

I've done this in two commits:
1. tidy up the previously-submitted tests.
2. add the new tests to cover more lines and methods in the class.

### CODE REVIEW

File: vjtop/src/test/java/com/vip/vjtools/vjtop/util/FormatsTest.java [hunk 1]
```diff
@@ -1,249 +1,124 @@
...
-        final String str = "!";
-        final int length = 1;
-        final int rightLength = 6;
-
-        // Act
-        final String retval = Formats.shortName(str, length, rightLength);
-
-        // Assert result
-        Assert.assertEquals("!", retval);
+        leftStr("!", -536_870_911);
```
Comment (JohnLBergqvist) [inline vjtop/src/test/java/com/vip/vjtools/vjtop/util/FormatsTest.java:45]: I usually keep the `Method is not expected to return due to exception thrown` comment after the call to the object under test.

File: vjtop/src/test/java/com/vip/vjtools/vjtop/util/FormatsTest.java [hunk 2]
```diff
@@ -1,249 +1,124 @@
...
     }
 
     @Test
-    public void toSizeUnitInputPositiveOutputNotNull() {
-
-        // Arrange
-        final Long size = 45_312L;
+    public void testToSizeUnitWithColor() {
+        final LongWarning warning = new LongWarning();
+        Assert.assertEquals("123", toSizeUnitWithColor(123L, warning));
```
Comment (JohnLBergqvist) [inline vjtop/src/test/java/com/vip/vjtools/vjtop/util/FormatsTest.java]: Couldn't you just inline `new LongWarning()` here?

**Review by JohnLBergqvist — CHANGES_REQUESTED** [review]
Format your PR message better.

File: vjtop/src/test/java/com/vip/vjtools/vjtop/util/FormatsTest.java (hunk 2 above)
Comment (petersear) [reply vjtop/src/test/java/com/vip/vjtools/vjtop/util/FormatsTest.java]: That would take the line over 80 chars.

File: vjtop/src/test/java/com/vip/vjtools/vjtop/util/FormatsTest.java (hunk 1 above)
Comment (petersear) [reply vjtop/src/test/java/com/vip/vjtools/vjtop/util/FormatsTest.java:45]: I was in two minds about that myself. I checked a few recent PRs and saw that people had left out the comment so did the same in the name of minimisation. I'm open to adding it back in though.

File: vjtop/src/test/java/com/vip/vjtools/vjtop/util/FormatsTest.java (hunk 2 above)
Comment (JohnLBergqvist) [reply vjtop/src/test/java/com/vip/vjtools/vjtop/util/FormatsTest.java]: Where has this 80 char limit come from? It's ridiculously short...

**Comment by petersear** [comment]
@JohnLBergqvist thanks, I've changed the PR text.
Happy to add the comment for the exception back in.

File: vjtop/src/test/java/com/vip/vjtools/vjtop/util/FormatsTest.java (hunk 2 above)
Comment (petersear) [reply vjtop/src/test/java/com/vip/vjtools/vjtop/util/FormatsTest.java]: From a previous PR comment and the style guide.

File: vjtop/src/test/java/com/vip/vjtools/vjtop/util/FormatsTest.java [hunk 3]
```diff
@@ -1,249 +1,77 @@
 package com.vip.vjtools.vjtop.util;
 
-import com.vip.vjtools.vjtop.WarningRule.LongWarning;
```
Comment (peterschrammel) [inline vjtop/src/test/java/com/vip/vjtools/vjtop/util/FormatsTest.java]: `Refactor previously submitted unit tests` -> `Refactor existing unit tests`

**Review by peterschrammel — APPROVED** [review]

File: vjtop/src/test/java/com/vip/vjtools/vjtop/util/FormatsTest.java (hunk 2 above)
Comment (paulkennethkent) [reply vjtop/src/test/java/com/vip/vjtools/vjtop/util/FormatsTest.java]: You could still inline and then split the assert over two lines?

**Review by paulkennethkent — APPROVED** [review]

**Review by sudo-jaa — APPROVED** [review]

**Review by JohnLBergqvist — APPROVED** [review]

**Pull request closed without merging**